tree.go

```diff
@@ -41,7 +43,7 @@ type (
 // Committer represents an object that is able to create the
 // commitment to a polynomial.
 type Committer interface {
-	CommitToPoly([]Fr, int) *Point
+	CommitToPoly([]Fr, int) Point
```

Comment:
This is aligning the change proposed [here](https://github.com/crate-crypto/go-ipa/pull/31).
The `CommitToPoly(...)` returns `Point` instead of `*Point` to avoid having a heap allocation every time this function is called.

Again, let's see how a merged #295  makes changes in this PR look like, and we can see if this change stills apply after that.